src/components/KRadio/KRadio.vue

```diff
@@ -10,106 +10,196 @@
       type="radio"
       @click="handleClick"
     >
-    <span
+    <div
+      v-if="type === 'card' && $slots.before"
+      class="k-radio-content-before"
+    >
+      <slot name="before" />
+    </div>
+    <template
       v-if="hasLabel"
-      class="k-radio-label"
     >
-      <slot>{{ label }}</slot>
-    </span>
+      <template v-if="type === 'default'">
+        <span class="k-radio-label">
+          <slot>{{ label }}</slot>
+        </span>
+      </template>
+      <template v-if="type === 'card'">
+        <span class="k-radio-label">
+          {{ label }}
+        </span>
+      </template>
+    </template>
     <div
       v-if="hasLabel && (description || $slots.description)"
       class="k-radio-description"
     >
       <slot name="description">{{ description }}</slot>
     </div>
+    <div
+      v-if="type === 'card' && $slots.after"
+      class="k-radio-content-after"
+    >
+      <slot name="after" />
+    </div>
   </label>
 </template>
 
-<script lang="ts">
-import { defineComponent, computed } from 'vue'
-
-export default defineComponent({
-  name: 'KRadio',
-  inheritAttrs: false,
-  props: {
-    /**
-     * Sets whether or not radio is selected
-     */
-    modelValue: {
-      type: [String, Number, Boolean, Object],
-      default: 'on',
-      required: true,
-    },
-    /**
-     * Overrides default label text
-     */
-    label: {
-      type: String,
-      default: '',
-    },
-    /**
-     * Overrides default description text
-     */
-    description: {
-      type: String,
-      default: '',
-    },
-    /**
-     * The value emitted from the radio on change if selected
-     */
-    selectedValue: {
-      type: [String, Number, Boolean, Object],
-      required: true,
+<script setup lang="ts">
+import { computed, useAttrs, useSlots } from 'vue'
+
+const props = defineProps({
+  /**
+   * Sets whether or not radio is selected
+   */
+  modelValue: {
+    type: [String, Number, Boolean, Object],
+    default: 'on',
+    required: true,
+  },
+  /**
+   * Overrides default label text
+   */
+  label: {
+    type: String,
+    default: '',
+  },
+  /**
+   * Overrides default description text
+   */
+  description: {
+    type: String,
+    default: '',
+  },
+  /**
+   * The value emitted from the radio on change if selected
+   */
+  selectedValue: {
+    type: [String, Number, Boolean, Object],
+    required: true,
+  },
+  type: {
+    type: String,
+    default: 'default',
+    validator: (value: string): boolean => {
+      return ['default', 'card'].includes(value)
     },
   },
```

Comment:
**Note for PR review:** besides this prop, all changes in `<script>` tag are due to switch to setup syntax.